### PR TITLE
Fix various typos

### PR DESCRIPTION
**Pull request by luzpaz**
## Main changes of this PR

Various typo fized

## Motivation and additional information

Typolessness 

## Author's checklist

* [ ] I added a changelog file with `make changelog` if there are user-observable changes since the last release.
* [ ] I ran `make format` to ensure everything is formatted correctly.
* [ ] I sticked to C++14 features.
* [ ] I sticked to CMake version 3.16.3.
* [ ] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [ ] Do you understand the code changes?

### CODE REVIEW

**Review by MakisH — APPROVED** [review]
Looks good to me, all changes are in messages and comments and the suggestions look correct.

File: extras/bindings/fortran/include/precice/SolverInterfaceFortran.hpp [hunk 1]
```diff
@@ -49,7 +49,7 @@ void precicef_initialize_(double *timestepLengthLimit);
 
 /**
  * Fortran syntax:
- * precicef_intialize_data()
+ * precicef_initialize_data()
```
Comment (DavidSCN) [inline extras/bindings/fortran/include/precice/SolverInterfaceFortran.hpp:52]: @MakisH not sure if we really want to change this. The function will be removed soon anyway, but it's still a bit odd to change it before.

Comment (MakisH) [reply extras/bindings/fortran/include/precice/SolverInterfaceFortran.hpp:52]: I don't see why not fixing the comment in any case. It does not rename the function anyway. Zero benefit / zero effort, so I see it as undefined behavior.

Comment (DavidSCN) [reply extras/bindings/fortran/include/precice/SolverInterfaceFortran.hpp:52]: Ah sorry, my fault, I thought it was the signature :see_no_evil:

**Pull request merged**